### ISSUE TITLE
Enable Github Pages

### ISSUE THREAD

**Issue by mgkuhn** [cambridgeuniversity/Project-Light#91]
Could you perhaps set under Settings on https://github.com/cambridgeuniversity/Project-Light/settings in the "GitHub Pages" section the "Source" option from currently "None" to "master branch", and thus enable GitHub Pages access here? (Do not choose a theme.)

This way, we could more easily browse the latest version of all the HTML documentation and example pages in this repository at https://cambridgeuniversity.github.io/Project-Light/ as demonstrated in my own fork at https://mgkuhn.github.io/Project-Light/

This would be especially useful in issues (like #70), to be able to provide a GitHub Pages link to quickly demonstrate to readers that a bug exists in the example pages of the latest version of this repository.

Note that enabling GitHub Pages access will make the current version of the files in this repository publicly available, but then this house-style was designed to be used on public-facing web pages, so that should not really be a problem.

**Comment by barneybrown** [comment]
My vote would be to move the whole thing over to GitLab and pick up from there.

Barney

**Comment by mgkuhn** [comment]
> My vote would be to move the whole thing over to GitLab and pick up from there.

While cloning the git repository history itself from github.com to gitlab.developes.cam.ac.uk is trivial, migrating all the users (how many do we have) and their issues (nearly 100) and comments for this project properly may not be so trivial: you would have to remap all the usernames that occur in issues. Few people use on github the CRSId that they have to use on https://gitlab.developers.cam.ac.uk/.

You probably would have to encourage users who have commented here to 
* create a gitlab.developers.cam.ac.uk account first, such that there is a name you can map them too, and
* encourage them to [ensure their GitLab account has an email address that matches the public email address of their GitHub user identity](https://docs.gitlab.com/ee/user/project/import/github.html).

Significant effort for what benefit? Project Light is not really justifiably a private project: most of its files are used on public websites anyway.

(P.S.: When replying via email on a github issue, be careful not to quote the entire message that you reply to, as that all gets added to the issue, see above. Better reply to an issue via the web interface, after clicking on the "view it on GitHub" link in the email footer.)

**Comment by mgkuhn** [comment]
@barneybrown Two things I just leant from @rjw57, who looks after the new UIS GitLab service at https://gitlab.developers.cam.ac.uk/ :
1. https://gitlab.developers.cam.ac.uk/ currently does **not** offer the GitLab equivalent of GitHub Pages, and adding that is not trivial the way they have set up GitLab at the moment. So if we want a place to browse the HTML pages in this repository as `application/html`, we should stay here on GitHub for the time being.
2. It is easy to migrate a GitHub project to https://gitlab.developers.cam.ac.uk/, via the "Import from GitHub" function that is offered to you when you create a new GitLab repository there. But first make sure that you go to https://gitlab.developers.cam.ac.uk/profile/account and that under "Social sign-in" you do connect your GitHub account with your UIS GitLab account. Also encourage all people who have created or commented on issues or pull requests for your project to do the same, such that their usernames can get converted correctly during the import, and they still receive notifications from the UIS GitLab service on any continued discussion of their issues there. (I've just done this.)

**Comment by rjw57** [comment]
> https://gitlab.developers.cam.ac.uk/ currently does not offer the GitLab equivalent of GitHub Pages, and adding that is not trivial the way they have set up GitLab at the moment.

Yes, sorry about that: https://gitlab.developers.cam.ac.uk/uis/devops/devhub/docs/wikis/FAQs#is-the-gitlab-pages-feature-available

**Comment by mvl22** [comment]
https://gitlab.developers.cam.ac.uk/uis/devops/devhub/docs/-/wikis/Changelog

"GitLab pages is now enabled."
